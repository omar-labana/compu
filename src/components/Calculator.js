import React from 'react';
import calculate from '../logic/calculate';
import '../App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';

class Calculator extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      total: null,
      next: null,
      operationName: null,
    };
  }

  handleClick = (buttonName) => {
    this.setState((state) => calculate(state, buttonName));
  }

  render() {
    const { total, next, operationName } = this.state;
    return (
      <main className="flex items-center justify-around">
        <h2 className="text-2xl">Letâ€™s do some math!</h2>
        <div className="calculator">
          <Display result={total} next={next} operationName={operationName} />
          <ButtonPanel clickHandler={this.handleClick} />
        </div>
      </main>
    );
  }
}

export default Calculator;
