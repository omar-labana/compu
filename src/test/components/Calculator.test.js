import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Calculator from '../../components/Calculator';

describe('Calculator Component', () => {
  const wrapper = shallow(<Calculator />);

  it('exists', () => {
    expect(wrapper.find('Calculator')).toBeTruthy();
  });

  it('has to match snap', () => {
    const calc = renderer.create(<Calculator />);
    expect(calc).toMatchSnapshot();
  });
});
