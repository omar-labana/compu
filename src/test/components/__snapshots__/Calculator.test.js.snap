// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component has to match snap 1`] = `
<main
  className="flex items-center justify-around"
>
  <h2
    className="text-2xl"
  >
    Let’s do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="bg-gray-400 h-24"
    >
      <h1
        className="text-white font-bold p-5 text-right text-3xl"
      >
        null
      </h1>
    </div>
    <div
      className="h-24 flex"
    >
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="bg-yellow-600 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
    </div>
    <div
      className="h-24 flex"
    >
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="bg-yellow-600 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="h-24 flex"
    >
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="bg-yellow-600 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="h-24 flex"
    >
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="bg-yellow-600 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="h-24 flex"
    >
      <button
        className="bg-gray-200 w-1/2 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="bg-gray-200 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="bg-yellow-600 w-1/4 border border-gray-300 text-2xl font-semibold"
        name="button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</main>
`;
